src/main/java/org/kaazing/netx/ws/internal/WebSocketExtensionFactory.java

```diff
@@ -40,11 +41,11 @@ private WebSocketExtensionFactory(Map<String, WebSocketExtensionFactorySpi> fact
      * {@link WebSocketExtensionFactorySpi}.
      *
      * @param name            the extension name
-     * @param formattedStr    RFC-3864 formatted string negotiated with the server including the extension name
+     * @param formattedStr    string representation of an extension formatted as HTTP request header
      *
      * @return WebSocketExtension   the parameterized extension
      */
-    public WebSocketExtensionSpi createExtension(String name, String formattedStr) {
+    public WebSocketExtensionSpi createExtension(String name, String formattedStr) throws IOException {
```

Comment:
Collapse to single argument as `extensionWithParams` and apply pattern matching to extract extension name and parameters here?